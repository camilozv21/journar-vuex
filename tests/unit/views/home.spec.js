import { shallowMount } from '@vue/test-utils'
import Home from '@/views/HomeView.vue'
import { test, suite } from "vitest";

suite('Pruebas en el Home View', () => {
    
    test('debe de renderizar el componente correctamente', ({ expect }) => {
        const wrapper = shallowMount( Home )
        expect( wrapper.html() ).toMatchSnapshot()
    })

    test('hacer click en un boton debe de redireccionar a no-entry', ({ expect }) => {
        
        let isPushCalled = false;
        const mockRouter = {
            push: () => { isPushCalled = true; }
        }

        const wrapper = shallowMount( Home, {
            global: {
                mocks: {
                    $router: mockRouter    
                }
            }
        })

        wrapper.find('button').trigger('click')

        expect( isPushCalled ).toBe(true)
    })
})