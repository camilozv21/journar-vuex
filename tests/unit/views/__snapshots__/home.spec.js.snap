// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Pruebas en el Home View > debe de renderizar el componente correctamente 1`] = `
"<div>
  <h1>Home</h1>
  <p>Welcome to the Home page</p><button class="btn btn-primary">boton</button><button class="btn btn-secondary">boton</button><button class="btn btn-success">boton</button>
</div>"
`;

exports[`Pruebas en el Home View debe de renderizar el componente correctamente 1`] = `
<div class="home"><img alt="Vue logo" src="">
  <h1>Hola Mundo</h1><button class="btn btn-primary">Primary</button><button class="btn btn-secondary">Secondary</button><button class="btn btn-success">Success</button>
</div>
`;
